frontend/src/Frontend/Examples/Chess/Main.hs

```diff
@@ -0,0 +1,189 @@
+        <> "draggable" =: "false"
+
+      backgroundColor (Just p') | p == p' = "yellow"
+      backgroundColor _                   = defaultColor p
+
+      defaultColor (Point i j) | (i + j) `mod` 2 == 0 = "grey"
+                               | otherwise            = "white"
+
+      translate (Just (ColoredPiece clr pp)) = translatePiece clr pp
+      translate _                 = $(static "chess/blank.svg")
+      translatePiece White King   = $(static "chess/kl.svg")
+      translatePiece White Queen  = $(static "chess/ql.svg")
+      translatePiece White Rook   = $(static "chess/rl.svg")
+      translatePiece White Bishop = $(static "chess/bl.svg")
+      translatePiece White Knight = $(static "chess/nl.svg")
+      translatePiece White Pawn   = $(static "chess/pl.svg")
+      translatePiece Black King   = $(static "chess/kd.svg")
+      translatePiece Black Queen  = $(static "chess/qd.svg")
+      translatePiece Black Rook   = $(static "chess/rd.svg")
+      translatePiece Black Bishop = $(static "chess/bd.svg")
+      translatePiece Black Knight = $(static "chess/nd.svg")
+      translatePiece Black Pawn   = $(static "chess/pd.svg")
+
+handleClick :: (WidgetState, Maybe GameState) -> Point -> (WidgetState, Maybe Move)
+handleClick (WidgetState highlight promotion, mBoard) new = (WidgetState newHighlight promotion, mMove) where
+  newHighlight = do
+    board <- mBoard
+    guard $ isNothing highlight
+    guard $ isNothing mMove
+    case board `indexGS` new of
+      Just (ColoredPiece color _) | color == gameState_turn board -> pure new
+      _                                                           -> Nothing
+  mMove = do
+    case highlight of
+      Just old -> do
+        moveFrom old
+      Nothing -> do
+        board <- mBoard
+        guard $ not $ config_selfCapture $ gameState_config board
+        [mv] <- pure $ do
+          old <- validSquares
+          Just mv <- pure $ moveFrom old
+          pure mv
+        pure mv
+  moveFrom old = do
+    board <- mBoard
+    let mv = Move old new (gameState_turn board) promotion
+    void $ move board mv
+    pure mv
+
+
+app
+  :: ( DomBuilder t m
+     , MonadFix m
+     , MonadHold t m
+     , PostBuild t m
+     )
+  => m ()
+app = localChessBoard
+
+localChessBoard
+  :: forall t m.
+     ( DomBuilder t m
+     , MonadFix m
+     , MonadHold t m
+     , PostBuild t m
+     )
+  => m ()
+localChessBoard = do
+    let initial = initialGS $ ChessConfig
+            -- TODO: Expose the feature or get rid of it
+            { config_selfCapture = False
+            }
+        mkMove mBoard mv = do
+          board <- mBoard
+          move board mv
+    rec
+        localBoard <- holdDyn (Just initial) $ attachWith mkMove (current localBoard) chessMove
+        chessMove <- chessBoard localBoard
+    pure ()
+
+chessBoard
+  :: forall t m.
+     ( DomBuilder t m
+     , MonadFix m
+     , MonadHold t m
+     , PostBuild t m
+     )
+  => Dynamic t (Maybe GameState)
```

Comment:
No `Maybe` as we discussed. If/when it goes back it can be separate as we discussed. 
```suggestion
  => Dynamic t GameState
```